source/dub/dub.d

```diff
@@ -275,16 +275,77 @@ class Dub {
 
 		const dubFolderPath = NativePath(thisExePath).parentPath;
 
+		// override default userSettings + localRepository if a $DPATH or
+		// $DUB_HOME environment variable is set.
+		bool overrideUserSettings;
+		bool overrideLocalRepository;
+		{
+			string dpathOverride = environment.get("DUB_HOME");
+			if (!dpathOverride.length) {
+				dpathOverride = environment.get("DPATH");
+				if (dpathOverride.length)
+					dpathOverride = (NativePath(dpathOverride) ~ "dub/").toNativeString();
+
+			}
```

Comment:
```suggestion
			string dubHome = environment.get("DUB_HOME");
			if (!dubHome.length) {
				if (auto dpath = environment.get("DPATH"))
					dubHome = (NativePath(dpath) ~ "dub/").toNativeString();
			}
```

IMO that makes more sense, no?

Reply:
we can't do the `if (auto ...` thing here because we need to check for length, otherwise someone running `env DPATH= dub` would wrongly get dub home set to `NativePath("") ~ "dub/"`